### PR TITLE
feat: ADR for PR reviews

### LINKED ISSUE #93: Short self-explanatory ADR for github review process using Prow
Write a short ADR so we don't have to explain this all the time.

- Explain OWNERS
- Explain `approvers`, `reviewers`
- Mandatory labels and chat-ops commands to assign them

### PR DESCRIPTION
**Pull request by tumido**
Resolves: #93

### CODE REVIEW

File: docs/adr/0016-pr-review.md
Comment (anishasthana) [inline docs/adr/0016-pr-review.md]: `person` isn't needed after `approver` here.

File: docs/adr/0016-pr-review.md
Comment (anishasthana) [inline docs/adr/0016-pr-review.md]: Shesheta -> Sesheta

Comment (anishasthana) [reply docs/adr/0016-pr-review.md]: `alligable` -> `eligible`

File: docs/adr/0016-pr-review.md
Comment (anishasthana) [inline docs/adr/0016-pr-review.md]: `starts` -> `start`

File: docs/adr/0016-pr-review.md
Comment (anishasthana) [inline docs/adr/0016-pr-review.md]: `needs` -> `need`

**Review by anishasthana — CHANGES_REQUESTED** [review]

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md]: Done, thank you :+1:

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md]: Done, thank you :+1:

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md]: Done, thank you :+1:

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md]: typo, should be: _Prerequisites_

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md:29]: If needed additional `OWNERS` files > If we need additional `OWNERS` files

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md:30]: "If needed additional `OWNERS` files" 
> 
"If needed, additional `OWNERS` files"

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md:37]: Is this how it's currently done? or is this how we want it done? the reviewer selection has seemed a bit random to me, I was under the impression it was just doing a round robin

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md]: We have an informal rule where the PRs that are large in nature, or introduce breaking changes require 2 lgtm's, and let it sit for at least one day giving people in a different time zone to get a chance to review it. Is this something we want to mention here?

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md:11]: Well..we don't have it in every repo yet, that in it self is also a problem. Maybe we should add it to all the repos  and as part of the template too?

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md:34]: The process looks like we can just copy paste it into the contributing.md files, is this what you had in mind?

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md:37]: What I meant with this is that the "selection" is based on `reviewers` list from `OWNERS` file. And which `OWNERS` file  is used is determined based on the modified files. Which `reviewers` are selected is pretty random.

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md]: yeah, we should

File: docs/adr/0016-pr-review.md
Comment (anishasthana) [reply docs/adr/0016-pr-review.md:37]: I thought it's round robin, but it can be "smarter" if  there's multiple owners files

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md:11]: Yeah, we probably should.

File: docs/adr/0016-pr-review.md
Comment (anishasthana) [reply docs/adr/0016-pr-review.md]: I think that's a good point -- we should probably codify that and mention it in all our contributing.md's

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md:34]: Yes, I think we can either link the ADR or copy it over there.

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [reply docs/adr/0016-pr-review.md:37]: Ahh gotcha

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md]: introduce > introduces

File: docs/adr/0016-pr-review.md
Comment (HumairAK) [inline docs/adr/0016-pr-review.md]: `PR is hold in review ` > `PR is held for review `

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md:11]: Actually, @HumairAK I don't feel like we should replicate content that much. Would you mind if I just transfer the contributing doc into blueprints as an ADR and then create a blank contributing doc in each repo linking to that contributing ADR?

Comment (HumairAK) [reply docs/adr/0016-pr-review.md:11]: yeah that's fine

File: docs/adr/0016-pr-review.md
Comment (larsks) [inline docs/adr/0016-pr-review.md:33]: Grammar nit for the above: "...is a list of GitHub users that are allowed to use **the** `/approve` chat-ops command"

File: docs/adr/0016-pr-review.md
Comment (larsks) [inline docs/adr/0016-pr-review.md]: Grammar nit: From **a** programming/coding point of view

But I would probably just drop that for:

> Reviewers assess the proposed changes and either provide feedback or respond with ...

I see below that you're trying to make some distinction between "programming" and "architectural", maybe, but I'm not sure that's germane; I think both the "reviewers" and "approver" are probably looking at a variety of aspects.

File: docs/adr/0016-pr-review.md
Comment (larsks) [inline docs/adr/0016-pr-review.md]: "Consequent" implies "caused by the approval", whereas "subsequent" just means "following". Which one do you mean here?

Comment (tumido) [reply docs/adr/0016-pr-review.md]: I've meant "subsequent", thank you :slightly_smiling_face:

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md]: Yeah.. I think those roles are best described in the upstream docs anyways. It's linked above when explaining the `OWNERS` file.

I've copied some of it from those docs and changed it here instead:

```diff
- Reviewers assess the changes from programming/coding point of view and either provide feedback
+ Reviewers assess the code for general code quality, correctness, sane software engineering, style. They either provide feedback
```

```diff
- Approvers evaluate architectural aspects of the PR and either provide feedback
+ Approvers evaluate architectural aspects of the PR, holistic acceptance criteria, including dependencies with other features, forwards/backwards compatibility, API and flag definitions. They either provide feedback
```

WDYT?

Comment (larsks) [reply docs/adr/0016-pr-review.md]: That seems fine, although there's a word missing in both phrases

> + Reviewers assess the code for general code quality, correctness, sane software engineering, **and** style. They either provide feedback

And:

> Approvers evaluate architectural aspects of the PR **and** holistic acceptance criteria, including dependencies with other features, forwards/backwards compatibility, API and flag definitions. They either provide feedback

**Review by anishasthana — APPROVED** [review]
/lgtm

File: docs/adr/0016-pr-review.md
Comment (tumido) [reply docs/adr/0016-pr-review.md]: Thank you, sir! :slightly_smiling_face: Updated! :+1:

**Comment by HumairAK** [comment]
/approve
:facepalm:

**Pull request merged**
